docs/weave-tutorial/05-messages.md

```diff
@@ -0,0 +1,109 @@
+---
+id: messages
+title: Messages
+sidebar_label: Messages
+---
+
+> [PR#2](https://github.com/iov-one/tutorial/pull/2): _Create msgs_
+
+Messages are requests for a change in the state, the action part of a transaction. They also need to be persisted (to be sent over the wire and stored on the blockchain), and must also be validated. They later are passed into [Handlers](https://godoc.org/github.com/iov-one/weave#Handler) to be processed and effect change in the blockchain state.
+
+## Messages vs. Transactions
+
+A message is a request to make change and this is the basic element of a blockchain. A transaction contains a message along with metadata and authorization information, such as fees, signatures, nonces, and time-to-live.
+
+A [Transaction](https://godoc.org/github.com/iov-one/weave#Tx) is fundamentally defined as anything persistent that holds a message:
+
+```go
+type Tx interface {
+    Persistent
+    // GetMsg returns the action we wish to communicate
+    GetMsg() (Msg, error)
+}
+```
+
+And every application can extend it with additional functionality, such as [Signatures](https://godoc.org/github.com/iov-one/weave/x/sigs#SignedTx), [Fees](https://godoc.org/github.com/iov-one/weave/x/cash#FeeTx), or anything else your application needs. The data placed in the Transaction is meant to be anything that applies to all modules, and is processed by a **Middleware**.
+
+A [Message](https://godoc.org/github.com/iov-one/weave#Msg) is also persistent and can be pretty much anything that an extension defines, as it also defines the Handler to process it. The only necessary feature of a Message is that it can return a `Path() string` which allows us to route it to the proper Handler.
```

Comment:
Reading this sentence one may think that the message defines the handler, which is not true. I think it is better to use short sentences and less context dependent words like `this` or `it`. Instead, always name what you describe to avoid misunderstandings.

---

>  The only necessary feature of a Message is that it can return a `Path() string` which allows us to route it to the proper Handler

Message should provide (not _return_) `Path() string` method.

Reply:
Nice catch